sys/vane/jael.hoon

```diff
@@ -1005,6 +1011,54 @@
     ^+  +>
     %-  cure(urb urb, sub sub, etn etn, sap sap, moz (weld (flop mos) moz))
     [hen our abet:(link:(burb our) ven)]
+  ::                                                    ::  ++wind:of
+  ++  wind                                              ::  rewind to snap
+    |=  [hen=duct our=@p block=@ud]
+    ^+  +>
+    ~&  %dripping
+    =/  old-qeu  snaps.sap
+    =:  snaps.sap       ~
+        count.sap       0
+        last-block.sap  0
+      ==
+    =^  snap=snapshot  +>.$
+      ?:  |(=(~ old-qeu) (lth block block-number:(need ~(top to old-qeu))))
+        [%*(. *snapshot latest-block.etn launch:contracts) +>.$]
+      |-  ^-  [snapshot _+>.^$]
+      ::  =^  [new-qeu=(qeu [block-number=@ud snap=snapshot]) snap=snapshot]  snaps.sap
+      =^  snap=[block-number=@ud snap=snapshot]  old-qeu
+        ~(get to old-qeu)
+      =:  count.sap       +(count.sap)
+          last-block.sap  block-number.snap
+          snaps.sap       (~(put to snaps.sap) snap)
+        ==
+      ?:  |(=(~ old-qeu) (lth block block-number:(need ~(top to old-qeu))))
+        [snap.snap +>.^$]
+      $
+    ~&  [%wind block latest-block.etn.snap (~(get by hul.eth.sub.snap) ~ben) ~(wyt by hul.eth.sub.snap)]
+    ::  Keep the following in sync with ++extract-snap:file:su
+    %=  +>.$
+      urb   urb.snap
```

Comment:
Shouldn't this just be restoring `eve.urb`, rather than all of `urb`? I'd guess that this would just be rolling back eth-related state.

So, this is probably more of a general snapshots question that I should've surfaced during the previous PR, but: are snapshots dealing with more than just eth- and eth-derived state? Why? I don't think we want to mess with misc (currently unused) jael state, do we?

Reply:
Yes, this should just be rolling back eth-derived state.

I have no idea what `pry.urb` is, so you're probably right.  I'll limit this to just `eve` within `urb`.  Do you see other state that shouldn't be stored in a snapshot?

Reply:
Probably `bal.sub` and `own.sub` too.

Reply:
Aren't at least `[jaw lyf tuf]:own.sub` downstream of ethereum?

Reply:
`own` has just become a somewhat messy grab-bag of state.

`tuf.own` *should* be downstream of ethereum, but is currently just set on boot and static thereafter. This needs to be the case for urbit/ethereum testnets, since the contract will (usually) not have the correct testnet domain in it.

`[jaw lyf]:own` are set on first boot, and will be subsequently changed if private keys are rotated with `%mint %jewel`. This is a holdover from the old %jael rite/promises/attestations/pki stuff. The intersection of that and the new ethereum stuff could really use a re-design pass, imo.

Reply:
Yes, `kyz.puk.sub` should be part of the snapshots I think. They're public keys, which should be downstream from Ethereum as well.

Reply:
The life and pass are, but the `live` bit in `+public` is not derived from ethereum (but from %meet moves sent by %ames upon neighboring with ship). We could put that in a separate structure if necessary.

Reply:
Now that `jaw` isn't in the snapshot, is there anything else that should be purged before one gives a snapshot to someone else (e.g. a star to a planet)?